crates/wascc-provider/src/lib.rs

```diff
@@ -393,6 +440,13 @@ impl<S: Store + Send + Sync> Provider for WasccProvider<S> {
     }
 
     async fn delete(&self, pod: Pod) -> anyhow::Result<()> {
+        let mut delete_key: i32 = 0;
+        for (key, val) in self.port_set.lock().unwrap().iter() {
```

Comment:
This could become a pretty expensive operation. You'll have to iterate over every taken port (which at minimum will be hundreds, but could be more). I would probably stick with the hash set and just iterate over all the ports from the pod and remove those from the hash set

Reply:
Nit: one thing you could do here too is just take the lock once

Reply:
agreed. Acquiring the lock once and re-using the lock on lines 444 and 449 would prevent excessive locking/unlocking operations.